src/mbgl/map/map_context.cpp

```diff
@@ -112,7 +112,12 @@ void MapContext::setStyleURL(const std::string& url) {
         if (res.status == Response::Successful) {
             loadStyleJSON(res.data, base);
         } else {
-            Log::Error(Event::Setup, "loading style failed: %s", res.message.c_str());
+            if (res.message.substr(res.message.length() - 3) == "404" &&
```

Comment:
I'll definitely defer to you on specific location (and this code is super-ugly right here), but it would be nice to add this check only once — especially since it's not platform dependent.